### DISCUSSION TITLE
react-navigation 7 rc version can't works well with webpack and electron.

### DISCUSSION THREAD

**Discussion by chinamcafee** [react-navigation/react-navigation#12050] [Q&A] [answered]
When I use react-navigation 6.x with webpack, webpack bundles every package correctly and my app works well in electron as follows:
![image](https://github.com/react-navigation/react-navigation/assets/3439961/1fc8a437-a6e4-4242-b40a-2db6e06f88b3)

But when upgrade react-navigation dependencies to 7.0 rc version, webpack can only bundle @react-navigation/bottom-tab, and my app can't works well:

<img width="1410" alt="image" src="https://github.com/react-navigation/react-navigation/assets/3439961/c4255213-902f-4b0e-b3e7-51b4fccf6520">

Why is this happening, how can this issue be resolved? Any help would be greatly appreciated.

**Comment by satya164** [comment]
Can you share a repo?

**Comment by chinamcafee** [comment]
@satya164 Thank you for your response. I have created a new repository. Please follow the README.md in the repository to reproduce the issue.

https://github.com/chinamcafee/react-navigation-electron-test

**Reply by satya164** [reply, marked as answer]
Thanks for the repro. Here is what I found:

- Webpack is resolving to the CommonJS build for React Navigation packages instead of the ES modules build.
- The CommonJS build has the `.cjs` file extension. However, the Webpack configuration doesn't support the `.cjs` file extension. Since it's a valid extension supported by NodeJS, it's a bug in the Webpack configuration.
- Since the `.cjs` extension isn't properly configured, the imports get handled by `asset/resource` loader - and the import returns the path to the generated bundle (which is a string) instead the module itself, which causes this error.

I wasn't able to repro this behavior in another app using `@expo/webpack-config` so it's likely something specific to this configuration.

To workaround this issue, add the following to the `webpack.config.js` to handle the `.cjs` extension:

```js
config.resolve.extensions.push('.cjs');
config.module.rules.forEach((rule: any) => {
  rule.oneOf?.forEach((r: any) => {
    if (r.type === 'asset/resource') {
      r.exclude.push(/\.cjs$/);
    }
  });
  rule.oneOf?.push({
    test: /\.cjs$/,
    use: {
      loader: 'babel-loader',
      options: {
        sourceType: 'unambiguous',
        babelrc: false,
        configFile: true,
      },
    }
  });
});
```

<img width="912" alt="image" src="https://github.com/react-navigation/react-navigation/assets/1174278/d304298d-3a11-4235-9681-576c4cfa902f">

Probably the proper fix would be to investigate why it's resolving to the CommonJS build instead of the ES module build. In addition, the Expo webpack config should probably be updated to properly handle `.cjs` file extension.

**Reply by chinamcafee** [reply]
Thank you for the solution you provided,  it indeed works.
Unfortunately, @expo/webpack-config is now in maintenance mode and no longer updated, and I haven't found a solution for using Metro Bundler with Electron. 

Perhaps I'll try to fork the @expo/webpack-config repository to investigate where the issue lies.

Thanks again for your help.
